test/test_core.py

```diff
@@ -5814,6 +5814,9 @@ def test_readdir(self):
   def test_readdir_empty(self):
     self.do_run_in_out_file_test('dirent/test_readdir_empty.c')
 
+  def test_readdir_unlink(self):
+    self.do_run_in_out_file_test('dirent/test_readdir_unlink.c')
```

Comment:
Side note, these tests should probably be in `other` - no reason to slow down CI by running them in all opt modes. But we can think about that separately, eventually.

Reply:
Sure, I'll move them in a follow-up.

Reply:
Yeah, the criteria are fuzzy to me too. We definitely want some testing in `core`. But probably 80% can go in `other`.

For JS, we do need to test multiple opt modes, so in general I agree, those can stay in core.

For C++, aside from some testing of multiple opt modes, most seems fine in `other`.